src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsTest.java

```diff
@@ -462,39 +469,44 @@ public void testAgentRefresh() throws Exception {
           "Creds should cache on master, pass to agent, and refresh agent from master once",
           credentialsLog,
           contains(
-              // (agent log added out of order, see below)
-              "Generating App Installation Token for app ID 54321", // 1
-              "Generating App Installation Token for app ID 54321", // 2
-              "Failed to generate new GitHub App Installation Token for app ID 54321: cached token is stale but has not expired", // 3
               // node ('my-agent') {
-              // checkout scm
-              // checkout scm
+              //   echo 'First Checkout on agent should use cached token passed via remoting'
+              //   git url: REPO, credentialsId: 'myAppCredentialsId'
+              "Generating App Installation Token for app ID 54321",
+              //   echo 'Multiple checkouts in quick succession should use cached token'
+              //   git ....
+              // (No token generation)
+              //   sleep
+              //   echo 'Checkout after token is stale refreshes via remoting - fallback due to
+              // unexpired token'
+              //   git ....
```

Comment:
Is there a diff link to the pre-#609 state for ease of review?

Reply:
https://github.com/jtnord/github-branch-source-plugin/compare/a14edd2fa7243d56726f62a88c746ce7d7de655a...JENKINS-69488#diff-03b8ddb1abb4051c64c1e2898e14bf79ff304a1fce3442fa3e789f0af96cdeab

Reply:
Thanks. I guess the non-obvious parts of the test diff are due to the addition of chronological sorting.